src/Microsoft.Health.Fhir.SqlServer/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs

```diff
@@ -233,13 +235,12 @@ public static IFhirServerBuilder AddSqlServer(this IFhirServerBuilder fhirServer
                 .Singleton()
                 .AsSelf();
 
-            // next block of services is for watchdogs
-            ////services.Add<SqlTransactionHandler>().Singleton().AsSelf();
-            ////services.Add<SqlConnectionWrapperFactory>().Singleton().AsSelf();
-            ////services.Add<SqlQueueClient>().Singleton().AsSelf();
-            services.Add<DefragWatchdog>().Singleton().AsSelf();
+            services.Add<DefragWatchdog>()
+                .Singleton()
+                .AsSelf()
+                .ReplaceService<INotificationHandler<StorageInitializedNotification>>();
```

Comment:
I think this is stopping the ExportJobWorker from being registered as a NotificationHandler. I'm not seeing the StorageInitialized message in the logs for the ExportJobWorker, but it is there for the SearchParameterDefinitionManager.

Reply:
ExportJobWorker gets registered in the OperationsModule file, which hasn't been touched by this PR. This is the same issue we saw in Prod a few months ago where it gets deregistered when a new service using ReplaceService.

Reply:
I'm adding a new extension to try and set us up for success here. Previously it did find and remove the first instance of the service. The new extension should check by ServiceType + Implementation